src/ecdsa/ot_based_ecdsa/test.rs

```diff
@@ -327,3 +327,132 @@ fn test_e2e_random_identifiers_with_rerandomization() -> Result<(), Box<dyn Erro
     run_sign_with_rerandomization(presign_result, public_key.to_element(), msg)?;
     Ok(())
 }
+
+fn split_even_odd<T: Clone>(v: Vec<T>) -> (Vec<T>, Vec<T>) {
+    let mut even = Vec::with_capacity(v.len() / 2 + 1);
+    let mut odd = Vec::with_capacity(v.len() / 2);
+    for (i, x) in v.iter().enumerate() {
+        if i % 2 == 0 {
+            even.push(x.clone());
+        } else {
+            odd.push(x.clone());
```

Comment:
If you don't go with the `partition` suggestion, you can skip the clones by using `into_iter` here

```suggestion
    for (i, x) in v.into_iter().enumerate() {
        if i % 2 == 0 {
            even.push(x);
        } else {
            odd.push(x);
```

Reply:
Suggestions by AI with one pass :) (not tested)

```rust
fn split_even_odd<T: Clone>(v: Vec<T>) -> (Vec<T>, Vec<T>) {
    v.into_iter()
        .enumerate()
        .partition::<Vec<_>, _>(|(i, _)| i % 2 == 0)
        .into_iter()
        .map(|(evens, odds)| {
            (
                evens.into_iter().map(|(_, x)| x).collect(),
                odds.into_iter().map(|(_, x)| x).collect(),
            )
        })
        .next()
        .unwrap_or_else(|| (Vec::new(), Vec::new()))
}
```

Or, more simply, using `fold`:

```rust
fn split_even_odd<T: Clone>(v: Vec<T>) -> (Vec<T>, Vec<T>) {
    v.into_iter().enumerate().fold((Vec::new(), Vec::new()), |(mut even, mut odd), (i, x)| {
        if i % 2 == 0 {
            even.push(x);
        } else {
            odd.push(x);
        }
        (even, odd)
    })
}
```